### PR TITLE
Oq update

### PR DESCRIPTION
**Pull request by poojamat**
*Issue #, if available:*

*Description of changes:*

Send snapshot when an update happens on an item in observe query complete item list which renders this item to not qualify the predicate.
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java [hunk 1]
```diff
@@ -311,22 +311,20 @@ public void querySavedDataWithNumericalPredicates() throws DataStoreException, I
...
         Consumer<DataStoreQuerySnapshot<Post>> onQuerySnapshot = value -> {
             if (count.get() == 0) {
+                assertEquals(4, value.getItems().size());
                 assertTrue(value.getItems().contains(savedModels.get(1)));
                 assertTrue(value.getItems().contains(savedModels.get(4)));
                 assertTrue(value.getItems().contains(savedModels.get(5)));
                 assertTrue(value.getItems().contains(savedModels.get(6)));
                 latch.countDown();
-            } else if (count.get() == 1) {
+            }else if (count.get() == 2) {
```
Comment (raphkim) [inline aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java:327]:
```suggestion
            } else if (count.get() == 2) {
```

File: aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java [hunk 2]
```diff
@@ -311,22 +311,20 @@ public void querySavedDataWithNumericalPredicates() throws DataStoreException, I
...
-        };
+        Consumer<Cancelable> observationStarted = value -> { };
         AtomicInteger count = new AtomicInteger(0);
         Consumer<DataStoreQuerySnapshot<Post>> onQuerySnapshot = value -> {
             if (count.get() == 0) {
+                assertEquals(4, value.getItems().size());
                 assertTrue(value.getItems().contains(savedModels.get(1)));
                 assertTrue(value.getItems().contains(savedModels.get(4)));
                 assertTrue(value.getItems().contains(savedModels.get(5)));
                 assertTrue(value.getItems().contains(savedModels.get(6)));
```
Comment (raphkim) [inline aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java:325]:
```suggestion
                List<Post> expected = Arrays.asList(
                    savedModels.get(1),
                    savedModels.get(4),
                    savedModels.get(5),
                    savedModels.get(6)
                );
                assertEquals(new HashSet<>(expected), new HashSet<>(value.getItems()));
```

**Review by raphkim — COMMENTED** [review]
You can use `NoOpAction` and `NoOpConsumer` in all places where you pass empty lambdas to clean up the code a bit. e.g. `NoOpAction.create()` can replace `() -> {}` for clarity.

File: aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java (hunk 1 above)
Comment (poojamat) [reply aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java]: Done.

**Comment by poojamat** [comment]
> You can use `NoOpAction` and `NoOpConsumer` in all places where you pass empty lambdas to clean up the code a bit. e.g. `NoOpAction.create()` can replace `() -> {}` for clarity.

**Comment by poojamat** [comment]
Done

**Review by raphkim — APPROVED** [review]

**Review by gpanshu — APPROVED** [review]

**Review by changxu0306 — APPROVED** [review]

**Review by eeatonaws — APPROVED** [review]

**Pull request merged**